src/bot/bot.ts

```diff
@@ -14,14 +14,10 @@ const NUM_REACT_MAP = {
 
 export class DiscordBot {
     private discord: Discord.Client;
-    private token: string;
     private spotify: SpotifyDAO;
-    private rifm: RifmDAO;
 
-    constructor(token: string, spotify: SpotifyDAO, rifm: RifmDAO) {
-        this.token = token;
+    constructor(spotify: SpotifyDAO) {
```

Comment:
This kinda runs opposite to how my logger PR treats these classes. I'm into having top level module variables for configuration for simplicities sake. Lemme know where your heads at and I'll refactor my PR

Reply:
With class dependencies like this you can either tightly couple them together (like you do here - this class *needs* a `SpotifyDAO` to be passed to it) or you can loosely couple things with interfaces. That would look something like this:

```
interface IMusicClient { // or ISpotify
  searchAlbums: (search: string) => Promise<Album[]> // or any if you'd like to stay general
}

export class DiscordBot {
  constructor(music: IMusicClient)
}
```

Just posting this here because I like talking about design, not requesting changes :D

Reply:
This is something I was wondering about.

Should bot.js be the thing that defines that interface since it knows what methods it needs? Or should the Spotify DAO actually just implement an interface we define else where?

Reply:
Either works. I don't have a strong preference and I think there are pros and cons to each. It is definitely a bit redundant to declare roughly the same interface in multiple files like I do in my logger PR. That's a good reason to centralize the interface at least